src/core/qgslabelingengine.cpp

```diff
@@ -85,12 +85,12 @@ QgsLabelingEngine::~QgsLabelingEngine()
 QList< QgsMapLayer * > QgsLabelingEngine::participatingLayers() const
 {
   QSet< QgsMapLayer * > layers;
-  Q_FOREACH ( QgsAbstractLabelProvider *provider, mProviders )
+  for ( QgsAbstractLabelProvider *provider : mProviders )
```

Comment:
why this doesn't require a qgis::as_const?

Reply:
Because the method is const (3 lines above)